Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Collections.Specialized;
+    using System.Globalization;
+    using System.Linq;
+
+    /// <summary>
+    /// WebHeaderCollection extension methods.
+    /// </summary>
+    internal static class WebHeaderCollectionExtensions
+    {
+        /// <summary>
+        /// For the given header collection, for a given header of name-value type, find the value of a particular key.
+        /// </summary>
+        /// <param name="headers">Header collection.</param>
+        /// <param name="headerName">Name of the header in the collection.</param>
+        /// <param name="keyName">Desired key of the key-value list.</param>
+        /// <returns>Value against the given parameters.</returns>
+        public static string GetNameValueHeaderValue(this NameValueCollection headers, string headerName, string keyName)
```

Comment:
We could have used something like [NameValeHeaderValue](https://msdn.microsoft.com/en-us/library/system.net.http.headers.namevalueheadervalue(v=vs.110).aspx) to process multi value header but that would have meant adding an additional nuget package dependency. Since we are sensitive to adding new nuget package dependencies - I chose to write the functionality myself.